ui/src/main/webapp/adminTools/webContextPolicyManager/reducer.js

```diff
@@ -31,7 +31,11 @@ const bins = (state = List(), { type, bin, bins, whitelistContexts, path, binNum
         return state.delete(binNumber)
       }
     case 'WCPM/EDIT_MODE_SAVE':
```

Comment:
Before editing a WCPM bin, a copy is saved off under a "beforeEdit" key. If the user presses "cancel", this copy is restored and overwrites any changes they had made to the bin. If the user presses "save" then the "beforeEdit" copy is deleted here, editing mode closes, and the changes are sent to the backend to be persisted.